server.py

```diff
@@ -146,6 +143,38 @@ def dashboard():
     contests = Contest.select().order_by(Contest.start_time)
     return bottle.template("dashboard.html", contests=contests)
 
+@app.get("/stats")
+@login_required
+def statistics():
+    sub_history = []
+    sub_stats_correct = []
+    sub_stats_total = []
+    sub_history_temp = Submission.select(Contest.code, ContestProblems.question, Submission.time, Submission.is_correct)\
+        .join(ContestProblems, on=(Submission.contestProblem == ContestProblems.id)) \
+        .join(Session, on=(Submission.user == Session.user))\
+        .switch() \
+        .join(Contest, on=(ContestProblems.contest == Contest.id)) \
+        .order_by(Submission.time.desc())
+    for sub in sub_history_temp.tuples():
+        sub_history.append(sub)
+    sub_stats_correct_temp = (
```

Comment:
since you already have all the submissions and their "correct/incorrect" values let's use that to construct this number instead of another database call?